MyString.java

```diff
@@ -36,7 +43,28 @@ public static int countChar(String str, char ch) {
      * @return true is str1 is a subset of str2, false otherwise
      */
     public static boolean subsetOf(String str1, String str2) {
-         //// Replace the following statement with your code
+
+        if (str1.length() > str2.length()) {
+            return false;
+        }
+
+        String strTest = str1;
+
+        for (int i = 0; i < str2.length(); i++) {
+            if (strTest.length() == 0) {

```

Comment:
condition is not needed. when it is the case never enters 2nd loop and return true after the loop